tree/treeplayer/test/regressions.cxx

```diff
@@ -173,3 +175,15 @@ TEST(TTreeReaderRegressions, IndexedFriend)
 
    gSystem->Unlink(fname);
 }
+
+// ROOT-4012 (JIRA)
+TEST(TTreeFormulaRegressions, ConstantAlias)
+{
+   TTree t("t","ti");
+   t.SetAlias("w","3");
+   TTreeFormula tf;
```

Comment:
`TTreeFormula` is not meant to be use this way and the result here is an invalid / unusable object.
To prevent this usage, we may want to apply this patch (and its consequences):
```
diff --git a/tree/treeplayer/inc/TTreeFormula.h b/tree/treeplayer/inc/TTreeFormula.h
index 17655ee4a2..ac3c2fb5b1 100644
--- a/tree/treeplayer/inc/TTreeFormula.h
+++ b/tree/treeplayer/inc/TTreeFormula.h
@@ -170,7 +170,7 @@ private:
    template<typename T> T GetConstant(Int_t k);
 
 public:
-   TTreeFormula();
+   TTreeFormula(TRootIOCtor*);
    TTreeFormula(const char *name,const char *formula, TTree *tree);
      ~TTreeFormula() override;
```
Albeit this may and may not break advanced user code.

Reply:
Maybe in a separate PR ? Or should I try it in this one?

Reply:
Yes. it should be in a separate PR.  Thanks.

Reply:
Is now https://github.com/root-project/root/pull/17708